Synergy/Controllers/TeamsController.cs

```diff
@@ -0,0 +1,76 @@
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Synergy.Authorization;
+using Synergy.Models;
+using Synergy.Services;
+using Permissions = Synergy.Authorization.Permissions;
+
+namespace Synergy.Controllers;
+
+[ApiController]
+[Route("/api/[controller]")]
+[Authorize]
+public class TeamsController : ControllerBase
+{
+    private readonly TeamsService _teamsService;
+    private readonly UsersService _usersService;
+
+    public TeamsController(TeamsService teamsService, UsersService usersService)
+    {
+        _teamsService = teamsService;
+        _usersService = usersService;
+    }
+
+    [AuthorizePermission(Permissions.CanRegisterTeam)]
+    [HttpPost("register-team")]
+    public async Task<IActionResult> RegisterTeam([FromBody] RegisterTeamBody body)
+    {
+        var team = body.Team;
+        var userId = body.UserId;
+        try
+        {
+            var teamId = await _teamsService.RegisterTeam(team);
+            await _usersService.AddTeamIdToUser(userId, teamId);
+            return Ok();
```

Comment:
Remember a Success code that doesn't return anything should be a 204 NoContent not a 200 OK